### PR TITLE
Feature/nick changed event

### PR DESCRIPTION
**Pull request by oreleraki**
Added event OnNickChanged which contains new event args with old and new nicknames.

This is to response of Issue #71

### CODE REVIEW

File: source/IrcDotNet/IrcClientMessageProcessing.cs [hunk 1]
```diff
@@ -27,8 +27,12 @@ protected internal void ProcessMessageNick(IrcMessage message)
 
             // Local or remote user has changed nick name.
             Debug.Assert(message.Parameters[0] != null);
-            sourceUser.NickName = message.Parameters[0];
-        }
+
+			string oldNickName = sourceUser.NickName;
```
Comment (alexreg) [inline source/IrcDotNet/IrcClientMessageProcessing.cs]: Use `var` wherever possible. Also, indentation messed up here, I think?

Comment (alexreg) [reply source/IrcDotNet/IrcClientMessageProcessing.cs]: Maybe it's a spaces vs. tabs issue, and that needs to be normalised? Check for that, and let's maybe add a `.editorconfig` file too.

File: source/IrcDotNet/IrcEventArgs.cs [hunk 2]
```diff
@@ -10,11 +10,38 @@
 
 namespace IrcDotNet
 {
-    /// <summary>
-    ///     Provides data for the <see cref="IrcClient.ChannelListReceived" /> event.
-    /// </summary>
-    /// <threadsafety static="true" instance="false" />
-    public class IrcChannelListReceivedEventArgs : EventArgs
+	/// <summary>
```
Comment (alexreg) [inline source/IrcDotNet/IrcEventArgs.cs]: Yeah, definitely a spaces vs. tabs issue here. I don't care which you want to go with, but let's have it consistent throughout the project.

File: source/IrcDotNet/IrcClientMessageProcessing.cs (hunk 1 above)
Comment (oreleraki) [reply source/IrcDotNet/IrcClientMessageProcessing.cs]: Ok, it's probably due to VisualStudioCode issue.
I'm using var on a non primitive types(with the exception of string) to be more explicit. If your Coding Standard are var everywhere i can fix that to be align. Although i'm big supporter of writing as Microsoft writes.

Comment (alexreg) [reply source/IrcDotNet/IrcClientMessageProcessing.cs]: Yeah, I prefer `var`everywhere, and AFAIK, that's the convention I used up to this point. Any half-decent IDE or editor like VS/VScode shows you the type anyway. :-)

**Comment by alexreg** [comment]
In that case, just close the PRs and rebase them over `release`? If they're bug fixes that is. If they're normal dev work, master is fine. Since you have no other active maintainer now, just commit directly to master/release without PRs. (If another one gets active again or comes aboard, then you can go back to the PR model to review each other's.)

**Comment by oreleraki** [comment]
@alexreg It doesn't allow me to do any merging.
Still writes
"Review required"
"At least 1 approving review is required by reviewers with write access."

**Comment by oreleraki** [comment]
@alexreg, Hi mate, i havn't heard from you and there still the issue of a "at least 1 approving review is required" on all PR i've submitted.
There is more information about this github feature in here: https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-reviews#required-reviews
I still have lots of enhancement i want to submit, so i hope it will get your attention soon.

**Comment by alexreg** [comment]
@oreleraki Sorry, I didn't get any notifications about it. PRs look good: just make sure you run "Format Document" on all the changed files first. I've given you Maintainer rights now, so you should be able to push directly to the branch (you may just have to do that rather than accepting your own PRs, I'm not sure).

**Comment by alexreg** [comment]
Approved all the PRs you requested from me. Hope that's in order now. In the future, just commit directly... until (when) you get a second maintainer, at least. :-)

**Pull request merged**